connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerConnector.java

```diff
@@ -77,6 +77,9 @@ public void initialize(ConnectorConfig connectorConfig) {
         try {
             this.config = connectorConfig.originalsStrings();
             log.debug("{} Initializing connector {} with config {}", this, connName, config);
+            if (isSinkConnector()) {
```

Comment:
Hmm, does validation here do much? Since we're not validating a bunch of other stuff here (i.e. all the validation of the base ConnectorConfig settings, anything defined by the connector itself, etc) there's still a lot of ways we could fail.

If we validate input before ever writing it to the config topic, it seems like we should catch most issues. I think having validation on starting a connector could potentially allow for better error messages, so may be worth doing. But could we just reuse the validation in AbstractHerder instead? For example, I think this method is only called by `Worker.startConnector`, which in turn is only called in two places by StandaloneHerder and DistributedHerder. Could we just add extra calls to `AbstractHerder.validateConnectorConfig` at those locations before we try to invoke this code?

Reply:
You're correct about where this method gets called within the main code, but there are repercussions to making an additional call to `AbstractHerder.validateConnectorConfig ` for the mocks used in tests. I can investigate this, but it will take more time.